src/ResourceManager/Network/Commands.Network/DdosProtectionPlan/RemoveAzureRMDdosProtectionPlanCommand.cs

```diff
@@ -0,0 +1,76 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AutoMapper;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDdosProtectionPlan", SupportsShouldProcess = true)]
+    public partial class RemoveAzureRmDdosProtectionPlan : NetworkBaseCmdlet
```

Comment:
@murilogr same comment about the `ResourceId` parameter set. Also, we should have an additional parameter set for `InputObject`, which is of type `PSDdosProtectionPlan`. This allows the users to pipe the result of `Get-AzureRmDdosProtectionPlan` to this cmdlet and remove the given DDoS protection plan(s).

Reply:
They already can pipe with the current object. I tried locally and it works:

```
PS D:\azure-powershell> $ddosProtectionPlan = New-AzureRmDdosProtectionPlan -ResourceGroupName ResourceGroupName -Name DdosProtectionPlanName -Location "West US"
PS D:\azure-powershell> $ddosProtectionPlan | Remove-AzureRmDdosProtectionPlan
PS D:\azure-powershell> Get-AzureRmDdosProtectionPlan
```

They probably can do that because of "ValueFromPipelineByPropertyName = true" in Name and ResourceGroupName. When I pipe from an object, it gets these values from the object.

Reply:
@murilogr right, but for consistency we would like the `ResourceId` parameter set to be the one that gets values from the object by property names, and the `InputObject` parameter set be the one that gets the value from the object